extension/js/api.js

```diff
@@ -0,0 +1,95 @@
+import $ from 'jquery';
+import endpoints from './api_endpoints.json';
+
+var api_url = (Jupyter.notebook_list || Jupyter.notebook).base_url + "api/packagemanager/";
+
+/*
+This function makes the API calls to the specified endpoint with the corresponding headers
+*/
+
+function api_call(endpoint, url, payload, success) {
```

Comment:
Maybe you should consider `failure` besides `success` no? To show an error to the user or something. But let's add it as an improvement for the next iteration.

Reply:
Sure. :+1: